File: examples/governance_static_go/compliance_static_go/domain/framework_requirement.go
package domain

type FrameworkRequirement struct {
	Article string `json:"article"`
	Section string `json:"section"`
	Description string `json:"description"`
	RiskCategory string `json:"risk_category"`
}

func NewFrameworkRequirement(article string, section string, description string, riskCategory string) (FrameworkRequirement, error) {
	v := FrameworkRequirement{
		Article: article,
		Section: section,
		Description: description,
		RiskCategory: riskCategory,
	}
	return v, nil
}

File: examples/governance/compliance_domain/lib/compliance_domain/regulatory_framework/framework_requirement.rb
module ComplianceDomain
  class RegulatoryFramework
    class FrameworkRequirement
      attr_reader :article, :section, :description, :risk_category

      def initialize(article:, section:, description:, risk_category:)
        @article = article
        @section = section
        @description = description
        @risk_category = risk_category
        check_invariants!
        freeze
      end

      def ==(other)
        other.is_a?(self.class) &&
          article == other.article &&
          section == other.section &&
          description == other.description &&
          risk_category == other.risk_category
      end
      alias eql? ==

      def hash
        [self.class, article, section, description, risk_category].hash
      end

      private

      def check_invariants!; end
    end
  end
end
